src/main/kotlin/com/neva/gradle/fork/config/properties/Property.kt

```diff
@@ -0,0 +1,68 @@
+package com.neva.gradle.fork.config.properties
+
+import java.net.MalformedURLException
+import java.net.URL
+import java.nio.file.InvalidPathException
+import java.nio.file.Paths
+
+class Property(private val definition: PropertyDefinition, private val prompt: PropertyPrompt) {
+
+  val name: String
+    get() = prompt.name
+
+  var value: String
+    set(newValue) {
+      prompt.value = newValue
+    }
+    get() = prompt.valueOrDefault ?: definition.defaultValue
+
+  val label: String
+    get() = if (required) "${prompt.label}*" else prompt.label
+
+  val type: PropertyType = definition.type
+
+  private val required: Boolean
+    get() = definition.required
+
+  fun validate(): Validator {
+    val validator = Validator(value)
+    if (required && value.isBlank()) {
+      validator.error("This property is required.")
+      return validator
+    }
+    if (shouldBeValidated()) {
+      when (definition.validator) {
+        null -> applyDefaultValidation(validator)
+        else -> validator.apply(definition.validator ?: {})
+      }
+    }
+    return validator
+  }
+
+  fun isInvalid() = validate().hasErrors()
+
+  private fun applyDefaultValidation(validator: Validator) = when (type) {
+    PropertyType.PATH -> validatePath(validator)
+    PropertyType.URL -> validateUrl(validator)
+    else -> {
```

Comment:
this empty else branch looks strange, when is not exhaustive, maybe you could just remove it

Reply:
why not just `else -> {}` // with no new line